### PR TITLE
Unexpected keyword argument 'bf16' : Modified train.py to handle exceptions appropriately

### PR DESCRIPTION
**Pull request by srivhash**
Added a custom function that checks if the enlisted parameters within kwargs are valid.
If the type error is thrown, then the function ignores that specific parameter and checks for the other parameters.

### CODE REVIEW

File: weak_to_strong/train.py [hunk 1]
```diff
@@ -17,6 +17,47 @@
...
+            problem_kwarg = str(e).split("'")[1]
+
+            # Remove the problematic keyword and retry
+            new_kwargs = {k: v for k, v in custom_kwargs.items() if k != problem_kwarg}
+            return try_initialize_model_with_kwargs(model_config, new_kwargs, num_labels, linear_probe)
+        else:
+            # If the error is not due to an unexpected keyword argument, re-raise it
+            raise
+
+def try_initialize_model_with_kwargs_2(model_config, custom_kwargs, num_labels=2, linear_probe=False):
```
Comment (WuTheFWasThat) [inline weak_to_strong/train.py]: the only difference is device_map="auto"?  you could just pass dict(**custom_kwargs, device_map="auto") and use a single function

File: weak_to_strong/train.py [hunk 2]
```diff
@@ -17,6 +17,47 @@
...
+        )
+    except TypeError as e:
+        # Check if the exception is due to an unexpected keyword argument
+        if "unexpected keyword argument" in str(e):
+            # Identify the problematic keyword
+            problem_kwarg = str(e).split("'")[1]
+
+            # Remove the problematic keyword and retry
+            new_kwargs = {k: v for k, v in custom_kwargs.items() if k != problem_kwarg}
+            return try_initialize_model_with_kwargs(model_config, new_kwargs, num_labels, linear_probe)
```
Comment (WuTheFWasThat) [inline weak_to_strong/train.py]: this is a bug i think?  recursing into the wrong function

**Review by WuTheFWasThat — CHANGES_REQUESTED** [review]

**Comment by srivhash** [comment]
Added the requested changes

**Comment by WuTheFWasThat** [comment]
using https://github.com/openai/weak-to-strong/pull/22 instead

**Pull request closed without merging**